### ISSUE TITLE
Type union comes out as "any"

### ISSUE THREAD

**Issue by danvk** [Microsoft/dtslint#179]
If I add this to `expectType.ts.lint`:

```ts
// $ExpectType "foo" | "bar"
declare let t: 'foo' | 'bar';
```

and run `npm run build && npm run test`, then I get the following error:

```
  // $ExpectType "foo" | "bar"
  declare let t: 'foo' | 'bar';

- ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ [Expected type to be:
-   "foo" | "bar"
- got:
-   any]
```

Somehow the type union gets converted to "any". Putting that same code in the TypeScript playground shows that TS understands the type correctly. I'm not sure what's going on here.

![image](https://user-images.githubusercontent.com/98301/50731142-678bec80-112b-11e9-83e3-6c2f5357ca33.png)

**Comment by danvk** [comment]
Here are a few variations that do work:

```ts
function test(t: 'foo' | 'bar') {
  t; // $ExpectType "foo" | "bar"
}

let t: 'foo' | 'bar';
t;  // $ExpectType "foo" | "bar"
```

So I guess you just can't do a type assertion on the line in which a variable is declared? It has to be on an expression?
